packages/host/app/lib/html-component.ts

```diff
@@ -0,0 +1,45 @@
+import { setComponentManager } from '@ember/component';
+import { capabilities } from '@ember/component';
+import { setComponentTemplate } from '@ember/component';
+import { htmlSafe } from '@ember/template';
+
+import { precompileTemplate } from '@ember/template-compilation';
+
+import { ComponentLike } from '@glint/template';
+
+class _HTMLComponent {
+  constructor(readonly html: string) {}
+}
+
+export type HTMLComponent = ComponentLike<{ Args: {} }>;
+export function htmlComponent(html: string): HTMLComponent {
```

Comment:
What are the advantages of this approach compared to the previous one with a normal component with triple curlies?

Reply:
This approach allows our block arguments to be simpler. It is trading off more complexity in the private implementation in favor of an easier to use public API.